_episodes/02-basics.md

```diff
@@ -366,11 +370,38 @@ learn about branches.
 
 ## Exercise: undo unstaged changes
 
-- Make some silly changes to `ingredients.txt` (e.g. add liquorice).
+- Make some changes to `ingredients.txt` (e.g. add liquorice).
 - Inspect the changes with `git status` and `git diff`.
 - Undo the unstaged changes with `git checkout ingredients.txt`.
 - Inspect the new situation with `git status` and `git diff`.
 
+
+## Different meanings of "checkout"
+
+In Git the command "checkout" updates files in the working tree. However, depending on the context it is used, the way the changes are made are different. For example it it could switch to a different branch to get the versions of all the file corresponding to that branch, or it could retrieve a different version of a single file, while everything else is kept unchanged.   
+
+Some examples;
+
+Switch to the less-salt branch
+```shell
+git checkout less-salt
+```
+Create a new branch called new_branch and switch to it,  all in one go.
+```shell
+git checkout -b new_branch
+```
+Rather than checking out a branch to work on it, check out a commit for inspection and discardable experiments.
+```shell
+git checkout --detach
```

Comment:
What will this command do precisely?

Reply:
If you had ask, then I should probably remove this.

This is something that can be used to get  'detached HEAD' state, to play with a file. e.g. change the Make file to use an additional compile option, for a test (e.g. O2 --> O3) , then checkout the Makefile afterwords,

 This as a  checkout  example here, Git may have better practices to achieve this 

git checkout --detach
M	example.doc
Note: checking out 'HEAD'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at 5cf5f7e... adding